cli/command/container/run.go

```diff
@@ -166,7 +166,11 @@ func runContainer(dockerCli command.Cli, opts *runOptions, copts *containerOptio
 		defer close()
 	}
 
-	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, copts.autoRemove)
+	// We don't need to wait if we're not attaching.
+	var statusChan <-chan int
+	if attach {
```

Comment:
I'm not sure if `attach` is right here, but I really don't understand `config.Attach*` at all anyway. Right now we might get a panic if you do `-a stdin` (or in some of the error paths) because we might read from a nil chan -- but I don't see a nice way of solving that.

A simpler way of stopping the issue behind this problem is to just drop the logging code from `waitExitOrRemoved`.

Reply:
Reading from a nil chain would block forever, not panic.
So we either need another `if attach` before reading below (so as not to block forever) or make a channel and close it...

I agree `waitExitOrRemoved` should not be logging.
It could return a channel can handle both exit status and error messages.

I think removing logging, or some way to gate logging, might be a better approach.